src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java

```diff
@@ -137,6 +142,29 @@ protected CheckinReceivePiecesHelper(Map<String, String> okapiHeaders, Context c
     processingErrors = new HashMap<>();
   }
 
+  public static String extractPoLineId(Object collection) {
+    Optional<String> poLineOptional;
+    if (collection instanceof CheckinCollection checkinCollection) {
+      poLineOptional = checkinCollection.getToBeCheckedIn().stream().map(ToBeCheckedIn::getPoLineId).findFirst();
+    } else if (collection instanceof ReceivingCollection receivingCollection)  {
+      poLineOptional = receivingCollection.getToBeReceived().stream().map(ToBeReceived::getPoLineId).findFirst();
+    } else {
+      throw new IllegalStateException("Unsupported checkIn or receiving model");
+    }
+    return poLineOptional.orElseThrow(() -> new IllegalStateException("PoLineId cannot be extracted from a checkIn/receiving model"));
+  }
+
+  public Future<Void> retrievePurchaseOrderPoLinePair(String poLineId, PiecesHolder holder, RequestContext requestContext) {
```

Comment:
Add a new method to fetch PO and POL before checkIn & receiving to be used during item status update later on